corehq/apps/sms/resources/v0_5.py

```diff
@@ -117,7 +118,7 @@ def _validate_users(self, user_data):
                 )
 
             self._validate_toplevel_fields(user_info, [
-                FieldDefinition('phone_number', True, basestring),
+                FieldDefinition('phone_number', True, six.string_types),
```

Comment:
Looks like the test failure may be related to these changes
```
ERROR: corehq.apps.sms.tests.test_registration:RegistrationAPITestCase.test_validation
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/mock/mock.py", line 1305, in patched
    return func(*args, **keywargs)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/tests/test_registration.py", line 612, in test_validation
    {'app_id': 123},
  File "/mnt/commcare-hq-ro/corehq/apps/sms/tests/test_registration.py", line 447, in make_api_post
    HTTP_AUTHORIZATION=auth, content_type='application/json')
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 548, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 350, in post
    secure=secure, **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 416, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 501, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 221, in wrapper
    response = callback(request, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 461, in dispatch_list
    return self.dispatch('list', request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 202, in dispatch
    return super(BaseUserSelfRegistrationResource, self).dispatch(request_type, request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/__init__.py", line 107, in dispatch
    return super(HqBaseResource, self).dispatch(request_type, request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 493, in dispatch
    response = method(request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 262, in post_list
    super(UserSelfRegistrationResource, self).post_list(request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1401, in post_list
    updated_bundle = self.obj_create(bundle, **self.remove_api_resource_names(kwargs))
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 250, in obj_create
    bundle = self.full_hydrate(bundle)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 224, in full_hydrate
    if not self.is_valid(bundle):
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1313, in is_valid
    errors = self._meta.validation.is_valid(bundle, bundle.request)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 148, in is_valid
    FieldDefinition('custom_registration_message', False, six.string_types),
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 103, in _validate_toplevel_fields
    {field_def.name: 'Expected type: {}'.format(field_def.type.__name__)}
AttributeError: 'tuple' object has no attribute '__name__'
-------------------- >> begin captured logging << --------------------
django.request: ERROR: Internal Server Error: /a/reg-api-test-1/api/v0_5/sms_user_registration/
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 221, in wrapper
    response = callback(request, *args, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 461, in dispatch_list
    return self.dispatch('list', request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 202, in dispatch
    return super(BaseUserSelfRegistrationResource, self).dispatch(request_type, request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/api/resources/__init__.py", line 107, in dispatch
    return super(HqBaseResource, self).dispatch(request_type, request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 493, in dispatch
    response = method(request, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 262, in post_list
    super(UserSelfRegistrationResource, self).post_list(request, **kwargs)
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1401, in post_list
    updated_bundle = self.obj_create(bundle, **self.remove_api_resource_names(kwargs))
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 250, in obj_create
    bundle = self.full_hydrate(bundle)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 224, in full_hydrate
    if not self.is_valid(bundle):
  File "/vendor/lib/python2.7/site-packages/tastypie/resources.py", line 1313, in is_valid
    errors = self._meta.validation.is_valid(bundle, bundle.request)
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 148, in is_valid
    FieldDefinition('custom_registration_message', False, six.string_types),
  File "/mnt/commcare-hq-ro/corehq/apps/sms/resources/v0_5.py", line 103, in _validate_toplevel_fields
    {field_def.name: 'Expected type: {}'.format(field_def.type.__name__)}
AttributeError: 'tuple' object has no attribute '__name__'
--------------------- >> end captured logging << --------------------
```

Reply:
definitely - just pushed an update